### PR TITLE
docs(apisix): wrote key-auth

### PR DESCRIPTION
**Pull request by An-DJ**

### CODE REVIEW

**Review by juzhiyuan — APPROVED** [review]

File: apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md [hunk 1]
```diff
@@ -3,4 +3,221 @@ title: Key Authentication
...
+    "key-auth": {}
+  }
+}'
+```
+
+### Validation
+
+Here are three kinds of validation:
+
+#### Without API Key
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md:87]: serial number is required

File: apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md [hunk 2]
```diff
@@ -3,4 +3,221 @@ title: Key Authentication
...
+Connection: keep-alive
+Server: APISIX/3.1.0
+```
+
+#### With Wrong Key
+
+```sh
+$ curl "http://127.0.0.1:9080/ip" -H 'apikey: wrong-key' -I
+```
+
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md:107]: Need a brief explanation

File: apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md [hunk 3]
```diff
@@ -3,4 +3,221 @@ title: Key Authentication
...
+### Validation
+
+Here are three kinds of validation:
+
+#### Without API Key
+
+```sh
+$ curl "http://127.0.0.1:9080/ip" -I
+```
+
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md:92]: Need a brief explanation

File: apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md [hunk 4]
```diff
@@ -3,4 +3,221 @@ title: Key Authentication
...
+Connection: keep-alive
+Server: APISIX/3.1.0
+```
+
+#### With Correct Key
+
+```sh
+$ curl "http://127.0.0.1:9080/ip" -H 'apikey: auth-one' -I
+```
+
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md:122]: Need a brief explanation

File: apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md [hunk 5]
```diff
@@ -3,4 +3,221 @@ title: Key Authentication
...
+## What is the Key Authentication
+
+Key authentication is a relatively simple but widely used authentication approach.
+
+1. Administrator adds an authentication key (API key) to the Route. 
+2. API consumers add the key to the query string or headers to authenticate their requests.
+
+## Create a Consumer
+
+First, let's create a consumer named `tom`, and enable the `key-auth` plugin with the key `auth-one`, which means all requests with the key `auth-one` are considered from `tom`.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md:48]: I don't like this key `auth-one`

File: apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md [hunk 6]
```diff
@@ -3,4 +3,221 @@ title: Key Authentication
...
+Connection: keep-alive
+Date: Thu, 09 Feb 2023 03:27:57 GMT
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Credentials: true
+Server: APISIX/3.1.0
+```
+
+### Disable Authentication
+
+Deleting the global rule to disable global authentication.
```
Comment (moonming) [inline apisix_versioned_docs/version-3.1.0/getting-started/key-authentication.md:136]: Disable or Delete? totally different

**Review by moonming — DISMISSED** [review]

**Review by shreemaan-abhishek — COMMENTED** [review]
LGTM

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**